File: programs-basic-java/src/test/java/com/local/numbertypes/PronicNumberTest.java
package com.local.numbertypes;

import static com.local.numbertypes.NumberType.PRONIC_NUMBER;

import java.util.List;

import org.junit.jupiter.api.Test;

class PronicNumberTest extends AbstractNumberTest {

	@Test
	void testPronicNumbers() {
		testNumbersTypeByFile(PRONIC_NUMBER, "pronicnumbers.txt");
	}

	@Test
	void testPronicNumbersByList() {
		// Pass the order as 5
		testNumbersTypeByList(PRONIC_NUMBER, List.of(2,6,12,20,30,42), 6);
	}
}

File: programs-basic-kotlin/src/test/kotlin/com/local/numbertypes/PronicNumberTest.kt
package com.local.numbertypes
import org.junit.jupiter.api.Test
class PronicNumberTest : AbstractNumberTest() {
	@Test
	fun testPronicNumbers() {
		testNumbersTypeByFile(com.local.numbertypes.NumberType.PRONIC_NUMBER, "pronicnumbers.txt")
	}

	@Test
	fun testPronicNumbersByList() {
		// Pass the order as 5
		testNumbersTypeByList(com.local.numbertypes.NumberType.PRONIC_NUMBER, java.util.List.of<Int>(2, 6, 12, 20, 30, 42), 6)
	}
}